### PR TITLE
Panda v1+ compatibility upgrdate. Minor test fixes

### PR DESCRIPTION
**Pull request by abohara**
Modifications to allow Brightwind library to be compatible with latest Pandas version (1+). The changes should allow the library to remain backward compatible with Pandas v0.24 as well.

Changes include:
- Panda v1+ has dropped support for 'M' and 'Y' in timeDelta function as the answer would vary depending on month and year. Instead this branch uses DateOffset to represent data frequency.
- All tests related to this change pass. There are some existing test conflict in the current dev branch unrelated to the new changes
- Some minor test conflicts are arising out of numerical floating point precision issues and there are minor fixes in the branch for those.

Suggestions:
- In future revision we should consider using pandas `infer_freq` to estimate data resolution. The current version is unable to determine frequency accurately for special cases such as bimonthly or quarterly.

### CODE REVIEW

**Review by BiancaMorandi — COMMENTED** [review]
@abohara Thanks for this very useful update! I have reviewed and added my comments. As you can see in my comments, I have tried to make some format changes and reword few doc strings or functions but when trying to commit my changes to this branch I get a `Permission to altosphere/brightwind.git denied to BiancaMorandi.`.  If you can give me the permission then I will commit my changes. Please let me know if you want to discuss any of my comments.

**Comment by abohara** [comment]
@BiancaMorandi Just seeing this. Sorry for the delay. I will likely not get to this for another week or two.  I will grant permission for you to make your edits if you are able to (please let me know if you are not able to modify)

**Comment by BiancaMorandi** [comment]
> @BiancaMorandi Just seeing this. Sorry for the delay. I will likely not get to this for another week or two. I will grant permission for you to make your edits if you are able to (please let me know if you are not able to modify)

@abohara ok, no problem. I can make all the updates then. I got permission to edit now. Thanks

File: setup.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def get_version(rel_path):
     classifiers=[
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
```
[the remark this answers is not in the document]
Comment (BiancaMorandi) [reply setup.py:65]: @abohara which python version did you use for your tests?

**Comment by BiancaMorandi** [comment]
@abohara I made all the updates I had initially requested in my comments. 

If you have time, can I ask you to run a test for `test_transform.py`? I run several tests with different Pandas versions but I would like to make sure that the updates I made are still giving you a pass.

Can I also ask you which python version did you use for your tests?

**Comment by abohara** [comment]
Sure @BiancaMorandi . Let me validate and will provide you updates on that including your question on pandas and python versions. Thanks for making the changes.

File: brightwind/transform/transform.py [hunk 2]
```diff
@@ -96,8 +111,13 @@ def _get_data_resolution(data_idx):
     averaging period.
 
     The algorithm finds the most common time difference between consecutive time stamps and returns the
-    most common time stamp.
+    most common time frequency. The expected frequency will be one of 'seconds', 'minutes', 'hours', 'days',
+    'weeks', 'months', 'years'.
 
+
+    (Deprecated). Pandas `timedelta` will no longer be used to support the date resolution as it cannot reliably
```
[the remark this answers is not in the document]
Comment (stephenholleran) [reply brightwind/transform/transform.py]: @BiancaMorandi does this mean this is not backward compatible anymore?

[the remark this answers is not in the document]
Comment (BiancaMorandi) [reply brightwind/transform/transform.py]: @stephenholleran No, this means that timedelta is not supported anymore by pandas version > 1. but pd.dateoffset is supported by previous versions of pandas. ie 0.25

File: brightwind/analyse/analyse.py [hunk 3]
```diff
@@ -783,16 +783,19 @@ def time_continuity_gaps(data):
     """
     indexes = data.dropna(how='all').index
     resolution = tf._get_data_resolution(indexes)
-    resolution_days = resolution / pd.Timedelta('1 days')
+    # If the data resolution is `1 month` or `1 year`, then the resolution will
```
Comment (stephenholleran) [inline brightwind/analyse/analyse.py:786]: @BiancaMorandi could you please add a "be" at the end of this line? ".....resolution will **be**....."

File: tests/test_transform.py [hunk 4]
```diff
@@ -170,19 +171,26 @@ def test_apply_wind_vane_dead_band_offset():
...
     for idx, period in enumerate(periods):
-        assert bw.transform.transform._freq_str_to_timedelta(period).total_seconds() == results[idx]
+        if type(bw.transform.transform._freq_str_to_dateoffset(period)) == pd.DateOffset:
+            # The data frequency is returned as a DateOffset. The time delta can be in seconds
+            # can be derived adding the date offset to an actual date.
+            assert (ref_date + bw.transform.transform._freq_str_to_dateoffset(period) - ref_date
+                    ).total_seconds() == results[idx]
+
+    # Check that data frequency is returned as a DateOffset.
+    assert type(bw.transform.transform._freq_str_to_dateoffset(period)) == pd.DateOffset
```
Comment (stephenholleran) [inline tests/test_transform.py:194]:
```
# Check that data frequency is returned as a DateOffset.
assert type(bw.transform.transform._freq_str_to_dateoffset(period)) == pd.DateOffset
```
@BiancaMorandi These 2 lines should be tabbed in one and be within the for loop.

**Comment by abohara** [comment]
@BiancaMorandi @stephenholleran  Apologies from my end for the delay ( it has been a hectic start ). The `test_transform.py` passed on my side. My tests are on Py 3.9.10. Also previously was tested on Py2.7.

**Comment by stephenholleran** [comment]
Hey @abohara,
I have been trying to merge this pull request but with no luck. GitHub is showing a conflict but I have no idea where. I was able to merge locally to check where the conflicts are but there were no conflicts. Any idea where the conflicts are or why the hell I can't merge this????

**Comment by abohara** [comment]
@stephenholleran Fixed. There was an issue in the change log

**Comment by stephenholleran** [comment]
Thanks @abohara, I can't understand why GitHub couldn't show me that conflict.

**Pull request merged**